README.md

```diff
@@ -13,13 +18,171 @@ idseq_dag examples/host_filter_dag.json
 
 ```
 
+` idseq_dag --help ` for more options
+
 ## Test
 
 ```
 cd idseq-dag; python3 -m unittest
 
 ```
 
+or `python3 -m unittest tests/<module_file> ` for testing individual modules.
+
+## DAG Execution Details
+
+### Composing an example dag json file
+There are four basic elements of an IdSeq Dag
+
+ - **output_dir_s3**: the s3 directory where the output files will be copied to.
+ - **targets**: the outputs that are to be generated through dag execution. Each target consists of a list of files that will be copied to *output_dir_s3*
+ - **steps**: the steps that will be exeucuted in order to generate the targets. For each step, the following attributes can be specified:
```

Comment:
Do the step correspond to the DAG's nodes?

Reply:
yes. each step generates a target. how would you phrase it?

Reply:
Thats fine. You did the matching steps/nodes in the beginning now.